content/blog/2023/2023-08-25-fairness-reweighing-preprocessor.md

```diff
@@ -0,0 +1,51 @@
+
+### New scoring metrics in Test and Score and Predictions:
+
+With the fairness addon and widgets that come with it, we also introduced four new scoring metrics for the Test and Score and Predictions widgets. We use these metrics to evaluate the fairness of model predictions. The metrics are:
+
+- Disparate Impact (DI): Measures the ratio of favorable outcomes for an unprivileged group to that of the privileged group. An ideal value of 1.0 means the ratio of favorable outcomes is the same for both groups.
+  - If DI < 1.0: The privileged group receives favorable outcomes at a higher rate than the unprivileged group.
+  - If DI > 1.0: The privileged group receives favorable outcomes at a lower rate than the unprivileged group.
+- Statistical Parity Difference (SPD): This is very similar to disparate impact. Instead of the ratio, it measures the difference in favorable outcomes between the unprivileged and the privileged groups. An ideal value for this metric is 0.0, which means that the difference in favorable outcomes is the same for both groups.
+  - If SPD < 0: The privileged group has a higher rate of favorable outcomes compared to the unprivileged group.
+  - If SPD > 0: The privileged group has a lower rate of favorable outcomes compared to the unprivileged group.
+- Average Odds Difference (AOD): This metric calculates the average gap between the true positive rates (correctly predicting a positive outcome) and false positive rates (incorrectly predicting a positive outcome) for both the privileged and unprivileged groups. A value of 0.0 indicates equal rates for both groups, signifying fairness.
+  - If AOD < 0: Indicates potential bias in favor of the privileged group.
+  - If AOD > 0: Indicates potential bias against the privileged group.
+- Equal Opportunity Difference (EOD): This metric is similar to the Average Odds Difference. It measures the difference in true positive rates between the unprivileged and the privileged group. An ideal value for this metric is 0.0, which means that the difference in true positive rates is the same for both groups.
+  - If EOD < 0: The privileged group has a higher true positive rate than the unprivileged group, suggesting possible bias.
+  - If EOD > 0: The privileged group has a lower true positive rate than the unprivileged group, suggesting potential bias against them.
+
+## Orange use case
+
+Now that we know all the required widgets and new fairness scoring metrics, let us explore a real-world example of using the reweighing widget as a preprocessor. We will use the [German credit dataset](http://archive.ics.uci.edu/dataset/144/statlog+german+credit+data) for this illustration. Like the previous blog, we will not use the As Fairness Data widget to select fairness attributes. Instead, we will use the default attributes, "sex" as the protected attribute and "male" as the privileged group, already defined in the dataset.
```

Comment:
The structure that first defines the widgets and then comes to the problem is, I think, not the one to use for the blog. First present the data, the problem, and then introduce methods as you go. There is also no need to introduce scores that you do not use in interpretation. As I see below you introduced scoring methods just fine, so the above section could probably be removed.